lib/commands/session.js

```diff
@@ -24,6 +25,8 @@ commands.createSession = async function (caps) {
     this.newCommandTimeoutMs = (this.caps.newCommandTimeout * 1000);
   }
 
+  this.resetOnUnexpectedShutdown();
```

Comment:
I realize I started this whole concept, but it's starting to get pretty confusing. I think we need some better comments or explanation about what this and other code is doing.